src/scripts/views/form/index.js

```diff
@@ -1,79 +1,117 @@
-import React       from 'react/addons';
-import Broadcaster from '../../components/broadcaster';
+import page  from 'page';
+import React from 'react';
+
+import Broadcaster     from '../../components/broadcaster';
+import FormData        from '../../views/form/form-map';
+import BroadcastAction from '../../actions/broadcast';
 
 /**
  *
  */
+
 export default React.createClass({
 	mixins: [ React.addons.LinkedStateMixin ],
-
 	propTypes: {
-		fields: React.PropTypes.arrayOf(React.PropTypes.shape({
-			type: React.PropTypes.oneOf([
-				'text', 'email', 'password'
-			]).isRequired,
-			name:     React.PropTypes.string.isRequired,
-			label:    React.PropTypes.string.isRequired,
-			title:    React.PropTypes.string,
-			pattern:  React.PropTypes.string,
-			required: React.PropTypes.bool
-		})),
-
-		secondary: React.PropTypes.shape({
-			submit:      React.PropTypes.func.isRequired,
-			action:      React.PropTypes.string.isRequired,
-			description: React.PropTypes.string
-		}),
-
-		help:   React.PropTypes.string,
-		submit: React.PropTypes.func.isRequired,
-		action: React.PropTypes.string.isRequired
-	},
-
-	getDefaultProps() {
-		return { fields: [ ], secondary: null }
+		formProfile: React.PropTypes.string.isRequired,
+		boardID: React.PropTypes.string,
+		accessCode: React.PropTypes.string
 	},
 
 	getInitialState() {
-		return this.props.fields.reduce((state, field) => {
+		return FormData.registerForm.fields.reduce((state, field) => {
 			state[field.name] = '';
 			return state;
 		}, {});
 	},
 
-	submitPrimary(event) {
-		this.props.submit(this.state);
-		return event.preventDefault();
+	checkPasswords(){
+		if(this.props.formProfile === 'registerForm' && this.state.passwordAgain !== '') {
+			return this.state.passwordAgain !== this.state.passwordRegister ?
```

Comment:
Leave the ternary "result" operators on their own lines in situations like these. However is this even possible?
```
return condition
  ? true
  : false
```